phalcon/collection.zep

```diff
@@ -24,13 +24,18 @@ class Collection implements \ArrayAccess, \Countable, \IteratorAggregate, \JsonS
 	/**
 	 * @var array
 	 */
-	private data = [];
+	protected data = [];
 
-	public function __construct(array! data = null) -> void
+	/**
+	 * @var array
+	 */
+	protected headers = [];
```

Comment:
Is it a common class for any other collections, right? Why do we need domain specific headers in the base class? I thought that `data` is enough for data storage. Why do we need `data` and `headers` here?

Reply:
How about HeadersBag or something like this?

Reply:
This is not the `headers` as you think of it. This array holds the case insensitive header -> case sensitive header map for faster searching and locating. The name of the internal array/property can be anything./ I just named it `headers`.

I will change it to arrayKeys or something.